revise blob tanks size

lcy03406 opened the pull request:
#### Summary
```SUMMARY: Mods "Revise blob fuel tanks"```

#### Purpose of change
+ Blob tanks were smaller than steel ones. It's weird since blob baskets are bigger than steel baskets.
+ As items, their contained sizes are bigger than volumes, I take it as blob's special ability. But the lowest leveled gelatinous tanks were the most efficient.
+ When folded, a blob tank including its contents shrinks to only 0.25L.

#### Describe the solution
+ modify the volume and contain size so the higher level is bigger and has better zip ratio.
+ change item names to match part names.
+ as suggested by @mlangsdorf, add a set of 'rigid' containers, very large, but cannot zip water.
+ name the new large tank simply 'tank'.
+ modify folded volume to match item size, which means liquids cannot be further zip.

Items:
+ gelatinous capsule: volume 5L (was 2.5L) contains 10L (was 7.5L)
+ oozing pod: volume 7.5L (not changed) contains 20L (was 15L)
+ gray cocoon: volume 10L (was 7.5L) contains 30L (was 25L)
+ gelatinous tank: volume and contains 60L
+ oozing tank: volume and contains 80L
+ gray tank: volume and contains 100L

Parts:
Foldable, but take the same space after folded.
+ gelatinous capsule: size 5L (was 10L), folded 5L (was 0.25L)
+ oozing pod: size 7.5L (was 20L), folded 7.5L (was 0.25L)
+ gray cocoon: size 10L (was 25L), folded 10L (was 0.25L)
+ gelatinous tank: 60L
+ oozing tank:  80L
+ gray tank: 100L

ZhilkinSerg commented:
You forgot enclosing quotation mark in summary line.

mlangsdorf commented:
why did you remove the looks_like? It wasn't harming anything.

lcy03406 commented:
@mlangsdorf 'looks_like' makes the tile completely transparent. I tried looks_like watercannon and it shows the item icon. So I guess it does work, but only items (not installed parts) are 'looks_like'able?

lcy03406 commented:
The blob fuel tanks are still not very useful, I'll tweak a bit.

lcy03406 commented:
OK I remove the position restriction that only one fuel source is allowed. Now on a tile you can install all 3 tanks, achieving 30+20+10 contain space, the same as a steel barrel.

mlangsdorf commented:
looks_like works on vehicle parts; I've got an APC with hddoors and hdstowboards and use looks_like to make it appear mostly sensible in chesthole. I can't see why it would work any different for a water cannon.

Please don't change the location of the tanks. Create a new base item with more capacity instead.

Even if you're going to change the location of the tanks, please delete the entire "location" line, don't partially comment it out.  That's a hack, and we're trying to clear the hacks out of blazemod, not add new ones in.

So here's what I'd prefer to see:
```
  {
    "id": "gloople_tank_large",
    "type": "CONTAINER",
    "copy-from": "gloople_tank",
    "name": "large gelatinous tank",
    "name_plural": "large gelatinous tanks",
    "weight": 13662,
    "volume": "60L",
    "price": 200,
    "contains": "60L"
  },
  {
    "id": "gloople_tank_large",
    "type": "vehicle_part",
    "copy-from": "gloople_tank",
    "name": "capsule (60L)",
    "item": "gloople_tank_large"
  },
  {
    "result": "gloople_tank_large",
    "type": "recipe",
    "category": "CC_OTHER",
    "subcategory": "CSC_OTHER_MATERIALS",
    "skill_used": "cooking",
    "difficulty": 6,
    "time": 15000,
    "reversible": true,
    "autolearn": true,
    "components": [ [ [ "gloople", 2 ] ] ]
  },
```

lcy03406 commented:
Well, I thought using tags instead of locations is the new preferred way. In 'vehicle::can_mount' comments says "Until we have an interface for handling multiple components with CARGO space, exclude them from being mounted in the same tile." I read it as ”we are willing to allow multiple components of same type but due to some user interface limitations we just cannot.“

In reality it's common to have multiple cargo space vertically overlap. On some big bus, you can have a luggage cabin under the seats level, have some space in and above the seats, and have a big bag above tied to the bus from outside. That's the same for liquid tanks. You won't say hay you can't install a water tank on your SUV because here is a fuel tank underneath the framework. Allowing multiple tanks on same tile is more of a feature rather than a hack.

Further more, the word "fuel_source" is not a location, it's a usage. Modern vehicle usually have tanks under the frame, but old trains have a cabin for coal and water storage. Who knows where do post cataclysm people install their fuel sources? In the game we have some strange shaped solar cars, so maybe their car design is not restricted on modern models.

And "fuel_source" is misleading, it's not about fuel, and it's not a source. A tank can store any liquid, and engines can draw fuel from any installed tanks. Both "fuel_source" location and "FUEL_TANK" tag only affects the user interface.

So if you are going to revise the vehicle system I highly recommend removing "furl_source" and limiting max installed components by their real position and size.

lcy03406 commented:
@mlangsdorf Large tanks added. And for the 'looks_like' problem, it turns out I 'looks_like'd a turret with wrong tile. I've changed it to a turret with non-empty tile.

mlangsdorf commented on data/mods/blazemod/blaze_blob_parts.json:
```diff
@@ -684,22 +693,31 @@
+    "name": "cocoon (30L)",
     "item": "gray_tank",
-    "location": "fuel_source",
+    "//location": "fuel_source",
```
This is not how CDDA does comments in JSON and this is not how CDDA removes a field from JSON. Also, standard vehicles can't put a 10L, 20L, and 60L tank on the same tile, so don't change how that works for blob parts.

mlangsdorf commented on data/mods/blazemod/blaze_blob.json:
```diff
@@ -584,20 +593,29 @@
+    "name": "gray tank",
+    "name_plural": "gray tanks",
+    "copy-from": "gray_tank",
+    "volume": 400,
```
Needs to mass more than a gray_tank - at least twice as much, since it has roughly 10x the volume and 4x the surface area.

mlangsdorf commented on data/mods/blazemod/blaze_blob.json:
```diff
@@ -431,20 +431,29 @@
+    "name_plural": "gelatinous tanks",
+    "copy-from": "gloople_tank",
+    "volume": 240,
+    "contains": 240
```
Needs to mass more than a gloople_tank - at least twice as much, since it has roughly 12x the volume and at least 4x the surface area.

mlangsdorf commented on data/mods/blazemod/blaze_blob.json:
```diff
@@ -860,10 +878,19 @@
+    "name_plural": "oozing tanks",
+    "copy-from": "oozle_tank",
+    "volume": 320,
+    "contains": 320
```
Needs to mass more than an oozle_tank - at least twice as much, since it has roughly 10x the volume and 4x the surface area.

mlangsdorf commented on data/mods/blazemod/blaze_blob_recipes.json:
```diff
@@ -234,6 +234,18 @@
+    "time": 15000,
+    "reversible": true,
+    "autolearn": true,
+    "components": [ [ [ "gloople", 1 ] ] ]
```
Should require 2 glooples.

mlangsdorf commented on data/mods/blazemod/blaze_blob_recipes.json:
```diff
@@ -285,6 +297,18 @@
+    "time": 15000,
+    "reversible": true,
+    "autolearn": true,
+    "components": [ [ [ "gray", 1 ] ] ]
```
Should require 2 grays.

mlangsdorf commented on data/mods/blazemod/blaze_blob_recipes.json:
```diff
@@ -404,6 +428,18 @@
+    "time": 15000,
+    "reversible": true,
+    "autolearn": true,
+    "components": [ [ [ "oozle", 1 ] ] ]
```
Should require 2 oozles.

lcy03406 commented:
Removed '//'. My fault. I thought I reverted all '//location' in the last commit.
I've convinced by you and won't make blazemod too weird, until you decide it's ok to install two tanks in a tile in the base game.

However for the recipe and mass:
+ A blob tank is already heavier than a normal metal tank, by 4 times.
+ Blobs multiply fast. Requiring two blobs does not change more. If you think it should take twice effort, we better set the time twice.
+ Each blob part is made by transforming a single blob. It can be as small as a seat belt, or as large as a wheel. A tank is not bigger than a frame so it's ok to follow the one blob one part rule.
+ Although multiplies fast, the first blob requires mutagen, and 60L metal tanks are abundant in cities. Blob tanks won't ruin early game balance.
+ Being fordable is the big advantage of blob cars. A capsule is better than a big tank in most blob car designs.

mlangsdorf commented:
Conservation of mass is what? Okay, you've convinced me.

Might be good, in the future, to establish the weight of a single blob. Then go through all the blob parts and set some sensible weights, then reset the recipes to use the appropriate amount of blob, but that's beyond the scope.
